identity/mock-dps-server/src/request.rs

```diff
@@ -0,0 +1,198 @@
+            device_id: Some(uuid.clone()),
+            status: Some("assigned".to_string()),
+            substatus: Some("initialAssignment".to_string()),
+            error_code: None,
+            error_message: None,
+            last_updated_date_time_utc: Some(now),
+            etag: Some("mock-dps-etag".to_string()),
+        };
+
+        if body.tpm.is_some() {
+            registration_state.tpm = Some(aziot_dps_client_async::model::TpmRegistrationResult {
+                authentication_key: "mock-dps-tpm-key".to_string(),
+            });
+        } else if self.headers.get("authorization").is_some() {
+            registration_state.symmetric_key = Some(
+                aziot_dps_client_async::model::SymmetricKeyRegistrationResult {
+                    enrollment_group_id: Some("mock-dps-enrollment-group".to_string()),
+                },
+            );
+        } else {
+            registration_state.x509 = Some(aziot_dps_client_async::model::X509RegistrationResult {
+                certificate_info: None,
+                enrollment_group_id: Some("mock-dps-enrollment-group".to_string()),
+                signing_certificate_info: None,
+            });
+        };
+
+        let operation_id = {
+            let registration = aziot_dps_client_async::model::RegistrationOperationStatus {
+                operation_id: uuid.clone(),
+                status: "assigned".to_string(),
+                registration_state: Some(registration_state),
+            };
+
+            let mut context = context.lock().unwrap();
+            context
+                .in_progress_operations
+                .insert(uuid.clone(), registration);
+
+            uuid
+        };
+
+        let response = aziot_dps_client_async::model::RegistrationOperationStatus {
+            operation_id,
+            status: "assigning".to_string(),
+            registration_state: None,
+        };
+
+        Response::json(hyper::StatusCode::OK, response)
+    }
+}
+
+struct OperationStatus {
+    _scope_id: String,
+    _registration_id: String,
+    operation_id: String,
+    _headers: std::collections::HashMap<String, String>,
+}
+
+impl DpsRequest for OperationStatus {
+    fn process(self, _body: Option<String>, context: &mut crate::server::DpsContext) -> Response {
+        let mut context = context.lock().unwrap();
+
+        match context.in_progress_operations.remove(&self.operation_id) {
+            Some(operation) => Response::json(hyper::StatusCode::OK, operation),
+            None => Response::not_found(format!("operation {} not found", self.operation_id)),
+        }
+    }
+}
+
+fn get_param(captures: &regex::Captures<'_>, name: &str) -> Result<String, Response> {
+    let value = &captures[name];
+
+    let value = percent_encoding::percent_decode_str(value)
+        .decode_utf8()
+        .map_err(|_| Response::bad_request(format!("bad {}", name)))?
+        .to_string();
+
+    Ok(value)
+}
+
+pub(crate) fn process_dps_request(
```

Comment:
Would it make sense to add TLS support to `http_common::Connector` instead of making an ad-hoc server like this? Or is this the only mock server we'll need?

Reply:
I think this is the only mock server we need. We probably don't need to mock IoT Hub since IS can function offline.

We can always add it later if needed.